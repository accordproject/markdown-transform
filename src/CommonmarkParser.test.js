/*
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// @ts-nocheck
/* eslint-disable no-undef */
'use strict';

const CommonmarkParser = require('./CommonmarkParser');
let parser = null;

// @ts-ignore
beforeAll(() => {
    parser = new CommonmarkParser();
});

test('can convert basic text', async () => {
    const markdownText = 'This is some text.';
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert italic text', async () => {
    const markdownText = 'This is *some* text.';
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert bold text', async () => {
    const markdownText = 'This is **some** text.';
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert bold italic text', async () => {
    const markdownText = 'This is ***some*** text.';
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert h1 to slate', async () => {
    const markdownText = '# Heading One';
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert h2', async () => {
    const markdownText = '## Heading Two';
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert h3', async () => {
    const markdownText = '### Heading Three';
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert h4', async () => {
    const markdownText = '#### Heading Four';
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert h5', async () => {
    const markdownText = '##### Heading Five';
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert h6', async () => {
    const markdownText = '###### Heading Six';
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert paragraphs', async () => {
    const markdownText = `This is first paragraph.
  
  This is second paragraph.`;
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert inline code', async () => {
    const markdownText = 'This is `inline code`.';
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert a link', async () => {
    const markdownText = 'This is [a link](http://clause.io).';
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert a thematic break', async () => {
    const markdownText = `This is 
  
  ---
  A footer.`;
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert a blockquote', async () => {
    const markdownText = `This is
  > A quote.`;
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert a multiline html block', async () => {
    const markdownText = `This is a custom
  
  <custom src="property">
  contents
  </custom>

  block.`;
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert an ordered list', async () => {
    const markdownText = `This is an ordered list:
1. one
1. two
1. three

Done.`;
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert an unordered list', async () => {
    const markdownText = `This is an unordered list:
* one
* two
* three

Done.`;
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert a single line html block', async () => {
    const markdownText = 'This is a <custom src="property"/> property.';
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert a code block', async () => {
    const markdownText = `\`\`\`
  this is a multiline
  code
  block.
  \`\`\``;
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert a custom code block', async () => {
    const markdownText = `\`\`\` <video src="https://www.youtube.com/embed/dQw4w9WgXcQ"/>
  this is a multiline
  code

  containing a newline

  which should be handled by the video plugin.
  \`\`\``;
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});

test('can convert a video block', async () => {
    const markdownText = `
  # Heading One
  
  <video src="https://www.youtube.com/embed/dQw4w9WgXcQ"/>
  `;
    const value = parser.parse(markdownText);
    expect(value).toMatchSnapshot();
});
