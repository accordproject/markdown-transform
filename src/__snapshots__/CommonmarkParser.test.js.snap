// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can convert a blockquote 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "column": 13,
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 156,
          "startTagPosition": 151,
          "text": "This is",
        },
      ],
      "position": 145,
      "startTagPosition": 135,
      "text": "
  ",
    },
    Object {
      "$class": "org.accordproject.commonmark.BlockQuote",
      "column": 15,
      "line": 6,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Paragraph",
          "column": 15,
          "line": 7,
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Text",
              "column": 12,
              "line": 8,
              "position": 230,
              "startTagPosition": 225,
              "text": "A quote.",
            },
          ],
          "position": 217,
          "startTagPosition": 207,
          "text": "
    ",
        },
      ],
      "position": 201,
      "startTagPosition": 189,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert a code block 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.CodeBlock",
      "column": 14,
      "line": 3,
      "position": 146,
      "startTagPosition": 135,
      "text": "  this is a multiline
  code
  block.
",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert a custom code block 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.CodeBlock",
      "column": 94,
      "info": "<video src=\\"https://www.youtube.com/embed/dQw4w9WgXcQ\\"/>",
      "line": 3,
      "position": 226,
      "startTagPosition": 135,
      "text": "  this is a multiline
  code

  containing a newline

  which should be handled by the video plugin.
",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert a link 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "column": 13,
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 156,
          "startTagPosition": 151,
          "text": "This is ",
        },
        Object {
          "$class": "org.accordproject.commonmark.Link",
          "column": 50,
          "destination": "http://clause.io",
          "line": 5,
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Text",
              "column": 12,
              "line": 6,
              "position": 235,
              "startTagPosition": 230,
              "text": "a link",
            },
          ],
          "position": 222,
          "startTagPosition": 177,
          "text": "
    ",
          "title": "",
        },
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 8,
          "position": 271,
          "startTagPosition": 266,
          "text": ".",
        },
      ],
      "position": 145,
      "startTagPosition": 135,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert a multiline html block 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "column": 13,
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 156,
          "startTagPosition": 151,
          "text": "This is a custom",
        },
      ],
      "position": 145,
      "startTagPosition": 135,
      "text": "
  ",
    },
    Object {
      "$class": "org.accordproject.commonmark.HtmlBlock",
      "column": 14,
      "line": 6,
      "position": 209,
      "startTagPosition": 198,
      "text": "  <custom src=\\"property\\">
  contents
  </custom>",
    },
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "column": 13,
      "line": 9,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 10,
          "position": 317,
          "startTagPosition": 312,
          "text": "block.",
        },
      ],
      "position": 306,
      "startTagPosition": 296,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert a single line html block 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "column": 13,
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 156,
          "startTagPosition": 151,
          "text": "This is a ",
        },
        Object {
          "$class": "org.accordproject.commonmark.HtmlInline",
          "column": 17,
          "line": 5,
          "position": 191,
          "startTagPosition": 179,
          "text": "<custom src=\\"property\\"/>",
        },
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 6,
          "position": 256,
          "startTagPosition": 251,
          "text": " property.",
        },
      ],
      "position": 145,
      "startTagPosition": 135,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert a thematic break 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "column": 13,
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 156,
          "startTagPosition": 151,
          "text": "This is",
        },
      ],
      "position": 145,
      "startTagPosition": 135,
      "text": "
  ",
    },
    Object {
      "$class": "org.accordproject.commonmark.ThematicBreak",
      "column": 20,
      "line": 6,
      "position": 206,
      "startTagPosition": 189,
    },
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "column": 13,
      "line": 7,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 8,
          "position": 231,
          "startTagPosition": 226,
          "text": "A footer.",
        },
      ],
      "position": 220,
      "startTagPosition": 210,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert a video block 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Heading",
      "column": 21,
      "level": "1",
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 164,
          "startTagPosition": 159,
          "text": "Heading One",
        },
      ],
      "position": 153,
      "startTagPosition": 135,
      "text": "
  ",
    },
    Object {
      "$class": "org.accordproject.commonmark.HtmlBlock",
      "column": 14,
      "line": 6,
      "position": 210,
      "startTagPosition": 199,
      "text": "  <video src=\\"https://www.youtube.com/embed/dQw4w9WgXcQ\\"/>",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert an ordered list 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "column": 13,
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 156,
          "startTagPosition": 151,
          "text": "This is an ordered list:",
        },
      ],
      "position": 145,
      "startTagPosition": 135,
      "text": "
  ",
    },
    Object {
      "$class": "org.accordproject.commonmark.List",
      "column": 65,
      "delimiter": "period",
      "line": 6,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Item",
          "column": 10,
          "line": 7,
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Paragraph",
              "column": 17,
              "line": 8,
              "nodes": Array [
                Object {
                  "$class": "org.accordproject.commonmark.Text",
                  "column": 14,
                  "line": 9,
                  "position": 312,
                  "startTagPosition": 307,
                  "text": "one",
                },
              ],
              "position": 297,
              "startTagPosition": 287,
              "text": "
      ",
            },
          ],
          "position": 279,
          "startTagPosition": 274,
          "text": "
    ",
        },
        Object {
          "$class": "org.accordproject.commonmark.Item",
          "column": 10,
          "line": 12,
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Paragraph",
              "column": 17,
              "line": 13,
              "nodes": Array [
                Object {
                  "$class": "org.accordproject.commonmark.Text",
                  "column": 14,
                  "line": 14,
                  "position": 397,
                  "startTagPosition": 392,
                  "text": "two",
                },
              ],
              "position": 382,
              "startTagPosition": 372,
              "text": "
      ",
            },
          ],
          "position": 364,
          "startTagPosition": 359,
          "text": "
    ",
        },
        Object {
          "$class": "org.accordproject.commonmark.Item",
          "column": 10,
          "line": 17,
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Paragraph",
              "column": 17,
              "line": 18,
              "nodes": Array [
                Object {
                  "$class": "org.accordproject.commonmark.Text",
                  "column": 14,
                  "line": 19,
                  "position": 482,
                  "startTagPosition": 477,
                  "text": "three",
                },
              ],
              "position": 467,
              "startTagPosition": 457,
              "text": "
      ",
            },
          ],
          "position": 449,
          "startTagPosition": 444,
          "text": "
    ",
        },
      ],
      "position": 268,
      "start": "1",
      "startTagPosition": 206,
      "text": "
  ",
      "tight": "true",
      "type": "ordered",
    },
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "column": 13,
      "line": 23,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 24,
          "position": 560,
          "startTagPosition": 555,
          "text": "Done.",
        },
      ],
      "position": 549,
      "startTagPosition": 539,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert an unordered list 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "column": 13,
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 156,
          "startTagPosition": 151,
          "text": "This is an unordered list:",
        },
      ],
      "position": 145,
      "startTagPosition": 135,
      "text": "
  ",
    },
    Object {
      "$class": "org.accordproject.commonmark.List",
      "column": 35,
      "line": 6,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Item",
          "column": 10,
          "line": 7,
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Paragraph",
              "column": 17,
              "line": 8,
              "nodes": Array [
                Object {
                  "$class": "org.accordproject.commonmark.Text",
                  "column": 14,
                  "line": 9,
                  "position": 284,
                  "startTagPosition": 279,
                  "text": "one",
                },
              ],
              "position": 269,
              "startTagPosition": 259,
              "text": "
      ",
            },
          ],
          "position": 251,
          "startTagPosition": 246,
          "text": "
    ",
        },
        Object {
          "$class": "org.accordproject.commonmark.Item",
          "column": 10,
          "line": 12,
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Paragraph",
              "column": 17,
              "line": 13,
              "nodes": Array [
                Object {
                  "$class": "org.accordproject.commonmark.Text",
                  "column": 14,
                  "line": 14,
                  "position": 369,
                  "startTagPosition": 364,
                  "text": "two",
                },
              ],
              "position": 354,
              "startTagPosition": 344,
              "text": "
      ",
            },
          ],
          "position": 336,
          "startTagPosition": 331,
          "text": "
    ",
        },
        Object {
          "$class": "org.accordproject.commonmark.Item",
          "column": 10,
          "line": 17,
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Paragraph",
              "column": 17,
              "line": 18,
              "nodes": Array [
                Object {
                  "$class": "org.accordproject.commonmark.Text",
                  "column": 14,
                  "line": 19,
                  "position": 454,
                  "startTagPosition": 449,
                  "text": "three",
                },
              ],
              "position": 439,
              "startTagPosition": 429,
              "text": "
      ",
            },
          ],
          "position": 421,
          "startTagPosition": 416,
          "text": "
    ",
        },
      ],
      "position": 240,
      "startTagPosition": 208,
      "text": "
  ",
      "tight": "true",
      "type": "bullet",
    },
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "column": 13,
      "line": 23,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 24,
          "position": 532,
          "startTagPosition": 527,
          "text": "Done.",
        },
      ],
      "position": 521,
      "startTagPosition": 511,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert basic text 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "column": 13,
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 156,
          "startTagPosition": 151,
          "text": "This is some text.",
        },
      ],
      "position": 145,
      "startTagPosition": 135,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert bold italic text 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "column": 13,
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 156,
          "startTagPosition": 151,
          "text": "This is ",
        },
        Object {
          "$class": "org.accordproject.commonmark.Emph",
          "column": 10,
          "line": 5,
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Strong",
              "column": 14,
              "line": 6,
              "nodes": Array [
                Object {
                  "$class": "org.accordproject.commonmark.Text",
                  "column": 14,
                  "line": 7,
                  "position": 212,
                  "startTagPosition": 207,
                  "text": "some",
                },
              ],
              "position": 197,
              "startTagPosition": 190,
              "text": "
      ",
            },
          ],
          "position": 182,
          "startTagPosition": 177,
          "text": "
    ",
        },
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 10,
          "position": 262,
          "startTagPosition": 257,
          "text": " text.",
        },
      ],
      "position": 145,
      "startTagPosition": 135,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert bold text 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "column": 13,
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 156,
          "startTagPosition": 151,
          "text": "This is ",
        },
        Object {
          "$class": "org.accordproject.commonmark.Strong",
          "column": 12,
          "line": 5,
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Text",
              "column": 12,
              "line": 6,
              "position": 197,
              "startTagPosition": 192,
              "text": "some",
            },
          ],
          "position": 184,
          "startTagPosition": 177,
          "text": "
    ",
        },
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 8,
          "position": 233,
          "startTagPosition": 228,
          "text": " text.",
        },
      ],
      "position": 145,
      "startTagPosition": 135,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert h1 to slate 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Heading",
      "column": 21,
      "level": "1",
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 164,
          "startTagPosition": 159,
          "text": "Heading One",
        },
      ],
      "position": 153,
      "startTagPosition": 135,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert h2 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Heading",
      "column": 21,
      "level": "2",
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 164,
          "startTagPosition": 159,
          "text": "Heading Two",
        },
      ],
      "position": 153,
      "startTagPosition": 135,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert h3 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Heading",
      "column": 21,
      "level": "3",
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 164,
          "startTagPosition": 159,
          "text": "Heading Three",
        },
      ],
      "position": 153,
      "startTagPosition": 135,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert h4 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Heading",
      "column": 21,
      "level": "4",
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 164,
          "startTagPosition": 159,
          "text": "Heading Four",
        },
      ],
      "position": 153,
      "startTagPosition": 135,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert h5 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Heading",
      "column": 21,
      "level": "5",
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 164,
          "startTagPosition": 159,
          "text": "Heading Five",
        },
      ],
      "position": 153,
      "startTagPosition": 135,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert h6 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Heading",
      "column": 21,
      "level": "6",
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 164,
          "startTagPosition": 159,
          "text": "Heading Six",
        },
      ],
      "position": 153,
      "startTagPosition": 135,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert inline code 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "column": 13,
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 156,
          "startTagPosition": 151,
          "text": "This is ",
        },
        Object {
          "$class": "org.accordproject.commonmark.Code",
          "column": 10,
          "line": 5,
          "position": 182,
          "startTagPosition": 177,
          "text": "inline code",
        },
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 6,
          "position": 211,
          "startTagPosition": 206,
          "text": ".",
        },
      ],
      "position": 145,
      "startTagPosition": 135,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert italic text 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "column": 13,
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 156,
          "startTagPosition": 151,
          "text": "This is ",
        },
        Object {
          "$class": "org.accordproject.commonmark.Emph",
          "column": 10,
          "line": 5,
          "nodes": Array [
            Object {
              "$class": "org.accordproject.commonmark.Text",
              "column": 12,
              "line": 6,
              "position": 195,
              "startTagPosition": 190,
              "text": "some",
            },
          ],
          "position": 182,
          "startTagPosition": 177,
          "text": "
    ",
        },
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 8,
          "position": 229,
          "startTagPosition": 224,
          "text": " text.",
        },
      ],
      "position": 145,
      "startTagPosition": 135,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;

exports[`can convert paragraphs 1`] = `
Object {
  "$class": "org.accordproject.commonmark.Document",
  "column": 48,
  "line": 2,
  "nodes": Array [
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "column": 13,
      "line": 3,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 4,
          "position": 156,
          "startTagPosition": 151,
          "text": "This is first paragraph.",
        },
      ],
      "position": 145,
      "startTagPosition": 135,
      "text": "
  ",
    },
    Object {
      "$class": "org.accordproject.commonmark.Paragraph",
      "column": 13,
      "line": 6,
      "nodes": Array [
        Object {
          "$class": "org.accordproject.commonmark.Text",
          "column": 10,
          "line": 7,
          "position": 227,
          "startTagPosition": 222,
          "text": "This is second paragraph.",
        },
      ],
      "position": 216,
      "startTagPosition": 206,
      "text": "
  ",
    },
  ],
  "position": 131,
  "startTagPosition": 84,
  "text": "
",
  "xmlns": "http://commonmark.org/xml/1.0",
}
`;
